5-app-infra/modules/env_base/main.tf

```
@@ -88,3 +95,116 @@ module "compute_instance" {
   instance_template     = module.instance_template.self_link
   resource_manager_tags = local.resource_manager_tags
 }
+
+
+module "confidential_instance_template" {
+  source  = "terraform-google-modules/vm/google//modules/instance_template"
+  version = "~> 13.0"
+
+  region     = var.region
+  project_id = local.env_project_id
+  subnetwork = local.subnetwork_self_link
+
+  source_image_project       = var.source_image_project != "" ? var.source_image_project : local.source_image_project
+  source_image               = var.source_image_family != "" ? var.source_image_family : local.source_image_family
+  machine_type               = var.confidential_machine_type
+  min_cpu_platform           = var.cpu_platform
+  enable_confidential_vm     = true
+  confidential_instance_type = var.confidential_instance_type
+
+  shielded_instance_config = {
+    enable_secure_boot          = true
+    enable_vtpm                 = true
+    enable_integrity_monitoring = true
+  }
+
+  metadata = {
+    tee-image-reference = var.docker_image_reference != "" ? var.docker_image_reference : local.default_tee_image_reference
+  }
+
+  service_account = {
+    email  = google_service_account.compute_engine_service_account.email
+    scopes = ["cloud-platform"]
+  }
+}
+
+module "confidential_compute_instance" {
+  source  = "terraform-google-modules/vm/google//modules/compute_instance"
+  version = "~> 13.0"
+
+  region                = var.region
+  subnetwork_project    = local.subnetwork_project
+  subnetwork            = local.subnetwork_self_link
+  num_instances         = var.num_instances
+  hostname              = var.confidential_hostname
+  instance_template     = module.confidential_instance_template.self_link
+  resource_manager_tags = local.resource_manager_tags
+}
+
+resource "google_service_account" "workload_sa" {
+  account_id   = "confidential-space-workload-sa"
+  display_name = "Workload Service Account for confidential space"
+  project      = local.env_project_id
+}
+
+resource "google_project_iam_member" "workload_sa_user" {
+  project = local.env_project_id
+  role    = "roles/iam.serviceAccountUser"
+  member  = "user:${data.google_client_config.default.account_id}"
```

```suggestion
  member  = var.confidential_space_workload_operator"
```

não existe `data.google_client_config` no codigo e vc tem que perguntar para o usuario